src/ray/util/BUILD

```diff
@@ -325,3 +325,16 @@ ray_cc_library(
         "@com_google_absl//absl/synchronization",
     ],
 )
+
+ray_cc_library(
+    name = "dup2_wrapper",
+    hdrs = ["dup2_wrapper.h"],
+    srcs = select({
+        "@platforms//os:windows": ["dup2_wrapper_windows.cc"],
+        "//conditions:default": ["dup2_wrapper_posix.cc"],
```

Comment:
What's the difference between this approach and the other approach the current codebase is using which is `ifdef`

Reply:
No difference, 
- `select` clause works at file level, `ifdef` works at source code level
- I use `select` here because implementation for two platforms are completely different and almost nothing to share